### DISCUSSION TITLE
(ktls) hardware offload using openssl 3?

### DISCUSSION THREAD

**Discussion by dvemail** [mitmproxy/mitmproxy#5314] [Q&A]
I have a 'smart' Mellanox card that has rx/tc tls hardware offload in the board's kernel.  Openssl 3.0.0 Alpha 10 works with this offload.  (I'm able to verify this using the Mellanox iperf_ssl tool) My question is how can I re-link mitmproxy or specify the library load order to use these new crypto libs instead of the default system openssl libs?

Any help will be greatly appreciated.

**Comment by mhils** [comment]
Let me say that mitmproxy is not built for high performance in general, but we use `cryptography` under the hood for everything OpenSSL. Hope that helps!
